lib/Db/CalendarShareRequest.php

```diff
@@ -37,28 +37,30 @@
 
 class CalendarShareRequest extends CalendarShareRequestBuilder {
 	/**
-	 * @param int $itemId
+	 * @param string $principalUri
+	 * @param string $uri
 	 *
 	 * @return Calendar
 	 * @throws CalendarDataNotFoundException
 	 */
-	public function getCalendarById(int $itemId): Calendar {
+	public function getCalendarByUri(string $principalUri, string $uri): Calendar {
 		$qb = $this->getCalendarSelectSql();
-		$qb->limitInt('id', $itemId);
+		$qb->limit('principaluri', $principalUri);
+		$qb->limit('uri', $uri);
 
 		return $this->getCalendarFromRequest($qb);
 	}
 
 
 	/**
-	 * @param int $itemId
+	 * @param int $calendar
```

Comment:
```suggestion
	 * @param int $calendarId
```
Using `$calendar` makes it seem like it would be a calendar object